### ISSUE TITLE
YYC: Compile error when trying to set image_number inside a struct or constructor method

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#6693]
### Description
image_number isn't a forbidden variable name to use in a struct, and setting it directly works fine. However, if you set it inside a method in a struct or constructor, the game will not compile in YYC, saying "assignment to read only variable". It compiles and works just fine in VM.

### Steps To Reproduce
```
global.struct = {
	image_number : 5,
	func : function() {
		image_number = 5;	
	}
}


function test() constructor {
	image_number = 5;
	static func = function() {
		image_number = 5;	
	}
}
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.6.1.160 Runtime v2024.6.1.208

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
Windows

[559bf6eb-8ae0-41b1-880f-ccf6f3d26f3c](https://api.gamemaker.io/api/github/downloads/559bf6eb-8ae0-41b1-880f-ccf6f3d26f3c)

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.800.0.593  Runtime v2024.800.0.617
